doc: make node(1) more consistent with tradition

strugee opened the pull request:
##### Checklist

- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

doc

##### Description of change

* Removed loads of whitespace
* Added traditional BUGS and AUTHORS sections
* Renamed ENVIRONMENT VARIABLES to ENVIRONMENT, as is traditional

In particular I'm not sure if everyone will be cool with the BUGS text - I personally think it's good but if anyone disagrees I'm not super attached to it. Also, I put `.\"`s instead of outright removing the newlines because I think it's (slightly) more readable, but I can also get rid of those.

bnoordhuis commented:
The "has many bugs" in the BUGS section is too nondescript to be useful, I'd change it to a REPORTING BUGS section with a link to the bug tracker.

I personally don't care for an AUTHORS section; it's not very relevant to readers and in our case it's misleading because Ryan hasn't worked on node for something like four years now.

Suggestion: a LICENSE section might be a good addition.

addaleax commented on doc/node.1:
```diff
@@ -149,61 +145,68 @@ Set v8's thread pool size which will be used to allocate background jobs.
+.\"
+.SH AUTHORS
+Written by Ryan Dahl and maintained by 1000+ contributors:
+.ur https://github.com/nodejs/node/blog/master/AUTHORS
```
Should this be “blob”?

strugee replied:
Nice catch! Fixed.

Fishrock123 commented:
I don't agree with the `./"` -- imo this will only make it harder for people less familiar to contribute to this. I suggest removing those.

Fishrock123 commented on doc/node.1:
```diff
@@ -149,61 +145,68 @@ Set v8's thread pool size which will be used to allocate background jobs.
+.ur https://github.com/nodejs/node/issues
+.\"
+.SH AUTHORS
+Written by Ryan Dahl and maintained by 1000+ contributors:
```
Effectively written by many others too. I would just link to the AUTHORS.

Fishrock123 commented on doc/node.1:
```diff
@@ -149,61 +145,68 @@ Set v8's thread pool size which will be used to allocate background jobs.
+.\"
+.SH BUGS
+Node.js has many bugs; they are tracked in GitHub Issues:
+.ur https://github.com/nodejs/node/issues
```
same, maybe just link the issue tracker? it's also linked below though....

imo, just remove

strugee commented:
@bnoordhuis @Fishrock123 changed the wording of the BUGS section to be more concise. If people still disagree with this I can just remove it entirely

I also changed AUTHORS to not mention Ryan, got rid of the `.\"`s and added a COPYRIGHT section

Fishrock123 commented on doc/node.1:
```diff
@@ -1,36 +1,30 @@
 .ft \\n(CF
 ..
-
-
```
Sorry, I meant not to touch these at all.

I _know_ it's not "perfectly syntactically correct" but it is far more readable.

In my experience the problems with rendering are negligible if any for the potential cost of people having a hard time contributing.

Fishrock123 commented on doc/node.1:
```diff
@@ -142,68 +113,65 @@ underscores (\fB_\fR).
+.SH BUGS
+Bugs are tracked in GitHub Issues:
+.ur https://github.com/nodejs/node/issues
+.SH AUTHORS
```
Add spaces between please (see above note)

Fishrock123 commented on doc/node.1:
```diff
@@ -142,68 +113,65 @@ underscores (\fB_\fR).
+Written and maintained by 1000+ contributors:
+.ur https://github.com/nodejs/node/blob/master/AUTHORS
+.SH COPYRIGHT
+Copyright Node.js contributors. All rights reserved. Node.js is available under the MIT license.
```
I don't think this correct -- all rights are not reserved, it is licensed under MIT.

strugee replied:
Honestly, I thought so too. But that's what LICENSE says. Does that need to be changed too?

strugee replied:
@Fishrock123 changed this despite the inconsistency with LICENSE.

Fishrock123 commented on doc/node.1:
```diff
@@ -142,68 +113,65 @@ underscores (\fB_\fR).
-
-.SH ENVIRONMENT VARIABLES
-
+.SH ENVIRONMENT
```
I think the previous wording is more obvious.

strugee commented:
All issues fixed, including the "All rights reserved" section (even though it's inconsistent with LICENSE).

jasnell commented on doc/node.1:
```diff
@@ -219,5 +231,5 @@ Mailing list:
 .ur "chat.freenode.net #node.js"
 
-IRC (node core development):
+IRC (Node core development):
```
s/Node/Node.js

jasnell commented on doc/node.1:
```diff
@@ -219,5 +231,5 @@ Mailing list:
 IRC (general questions):
 .ur "chat.freenode.net #node.js"
```
Nit: Given that `#node.js` is not *yet* considered an official resource managed by the project, a note indicating it's unofficial status might be worthwhile.

Fishrock123 commented on doc/node.1:
```diff
@@ -202,8 +202,20 @@ Setting this will void any guarantee that stdio will not be interleaved or
+.SH BUGS
+Bugs are tracked in GitHub Issues:
+.ur https://github.com/nodejs/node/issues
+.SH AUTHORS
```
could please add spaces here? looks good other than that

strugee commented:
:thumbsup: all issues fixed; this is ready to merge

strugee commented on doc/node.1:
```diff
@@ -202,8 +202,25 @@ Setting this will void any guarantee that stdio will not be interleaved or
+.SH COPYRIGHT
+Copyright Node.js contributors. Node.js is available under the MIT license.
+
+Node.js also includes external libraries that are available under a variety of licenses. See
```
Fixed

jasnell commented:
Landed in 90cd39f. Thank you!

strugee commented:
Thanks very much! :thumbsup:
